app/policies/wiki_policy.rb

```diff
@@ -1,19 +1,34 @@
 class WikiPolicy < ApplicationPolicy
   
-  class Scope < Scope
-
+  class Scope
+    attr_reader :user, :scope
+ 
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+ 
     def resolve
-      # if admin, all
-      # if guest, public
-      # if premium, your own + public + collaborating
-      # if standard, yours + public +collaborating
       wikis = []
-      scope.all.each do |wiki|
-        if user.present? && (user.admin? || !wiki.private || wiki.user == user) #wiki.users.include?(user)
-          wikis << wiki
+      if user.role == 'admin'
+        wikis = scope.all # if the user is an admin, show them all the wikis
+      elsif user.role == 'premium'
+        all_wikis = scope.all
+        all_wikis.each do |wiki|
+          if !wiki.private? || wiki.user == user || wiki.collaborators.include?(user)
```

Comment:
wiki.collaborators.include?(user) won't work because `wiki.collaborators` returns a collection of collaborator objects. We're then looking for a user object inside of this collection. Instead, with the association we talked about above, we could do `wiki.users.include?(user)` and now we're apples to apples.